base/strings/string.jl

```diff
@@ -176,9 +176,9 @@ is_valid_continuation(c) = c & 0xc0 == 0x80
 
 ## required core functionality ##
 
-@propagate_inbounds function iterate(s::String, i::Int=firstindex(s))
-    i > ncodeunits(s) && return nothing
-    b = codeunit(s, i)
+@inline function iterate(s::String, i::Int=firstindex(s))
+    (i % UInt) - 1 < ncodeunits(s) || return nothing
```

Comment:
```suggestion
    (i % UInt) <= ncodeunits(s) || return nothing
```
Wouldn't this be simpler? Does that generate worse code?

Reply:
Differs for `i==0`, no?

Reply:
Ah yes, I see.